### ISSUE TITLE
Backprojection_Loss label generation

### ISSUE THREAD

**Issue by JZlearnCode** [wvangansbeke/LaneDetection_End2End#57]
Hello, may I ask how to generate the label for the Backprojection_Loss training? I calculated the parameters as y=ax^2+bx+c for the Tusimple dataset, but the parameters were different from the Curve_parameters.json file. The x y coordinates are normalized to [0-1]. I used numpy.fitpoly.  Thanks a lot!

**Comment by wvangansbeke** [comment]
Hi @JZlearnCode,
It depends how high you go when you do the fit in bird's eye view. Some points can be mapped to infinity depending on the homography transformation. You can actually find the maximum height value mathematically, but I think I just took 0.7 in the normalized grid. 
Normally it shouldn't matter that much though. If you visualize the lines, they should be fairly similar in most cases.

Best, 
Wouter

**Comment by JZlearnCode** [comment]
Hi @wvangansbeke , Thank you for helping! 
May I ask 2 more questions?
1. Why the line existence (`line_lst`) has ground truth as 0, 1, 2 and softmax cross-entropy, instead of using 0 and 1 as binary output with logistic loss? 
```
https://github.com/wvangansbeke/LaneDetection_End2End/blob/9ab94b04dd0e7be393a8eb9960e4e3c92798ca8f/Birds_Eye_View_Loss/Dataloader/Load_Data_new.py#L117
```
During prediction, the `outputs_line` here are the index for maximun value (argmax). Since the output matrix for lines `outputs_line` has dimension batch_size x 3 x num_lines, the predicted argmax will have values 0, 1, 2. 

How was the line existence label generated in this link?
```
https://github.com/wvangansbeke/LaneDetection_End2End/blob/master/Birds_Eye_View_Loss/Labels/label_new.json
```

2. Why are the curve parameters different for Birds eye view v.s. backprojection loss? 
Birds eye view loss
```
"raw_file": "clips/0313-1/6040/20.jpg", "run_time": 20, "poly_params": [[-0.0007490195849774594, 0.04801345164633508, 0.4660409005249273], [-0.00011062192107513767, -0.020858596992042733, 0.5695067741417277], [-0.0028246980217723466, 0.12772983455234116, 0.3673634221909625], [0.001013433584937072, -0.10457649566360844, 0.6928532720714268]]
```
Back projection loss: 
```
"raw_file": "clips/0313-1/6040/20.jpg", "run_time": 20, "poly_params": [[-0.0006496997648889662, 0.032857701032923604, 0.4660359613854808], [-0.00023303610063694447, 0.010365525020821735, 0.5694987884336548], [-0.0018038300605271547, 0.06805293006049966, 0.36738668756377474], [0.00044512423977775654, -0.01792813981837305, 0.6928063035918499]]}
```

**Comment by wvangansbeke** [comment]
Hi @JZlearnCode,

Sorry for the late reply. I think the answers can be found in previous issues . Feel free to reopen if you're still looking for answers.

Best,
Wouter
